tests/integ/modin/frame/test_head_tail.py

```diff
@@ -71,3 +71,28 @@ def test_empty_dataframe(n, empty_snowpark_pandas_df):
         comparator=eval_result_and_query_with_no_join,
         check_column_type=False,
     )
+
+
+@pytest.mark.parametrize(
+    "ops",
+    [
+        lambda df: df.head(),
+        lambda df: df.iloc[1:100],
+        lambda df: df.iloc[1000:100:-1],
+    ],
+)
+@sql_count_checker(query_count=6)
+def test_head_efficient_sql(session, ops):
```

Comment:
do we have a follow up ticket to add head and iloc to our microbenchmarks?

Reply:
https://snowflakecomputing.atlassian.net/browse/SNOW-1731614